Update ha sequencer docs to use helm charts (#2273)
OffchainLabs/arbitrum-docs: arbitrum-docs/run-arbitrum-node/sequencer/high-availability-sequencer-docs.mdx

  alt="send to active disable"
  className="img-600px"
/>

## Using Helm Charts for Deployment

> We recommend using the [Offchain Labs Community Helm Charts](https://github.com/OffchainLabs/community-helm-charts) for deploying your high availability sequencer setup. These charts provide pre-configured templates for all the necessary components and make it easier to maintain your deployment. Base configuration values are provided in the examples below, however you should adjust them to fit your specific needs and utilize values files for production deployments.

Reviewer: ```suggestion
We recommend using the [Offchain Labs community Helm charts](https://github.com/OffchainLabs/community-helm-charts) to deploy your high-availability sequencer setup. These charts provide pre-configured templates for all the necessary components and make it easier to maintain your deployment. Base configuration values are provided in the examples below. However, you should adjust them to fit your specific needs and use values files for production deployments.
```

- Configure to listen to feed outputs from all sequencers
- All other components connect to these relays instead of directly to the sequencers
- Minimize direct load on sequencer nodes
Deploy sequencer relays using the relay helm chart:

> ```console

Reviewer: ```suggestion
```shell
```

helm install sequencer-relay offchainlabs/relay \
  --set replicaCount=2 \
  --set configmap.data.chain.id=<your-chain-id> \
  --set configmap.data.node.feed.input.url=ws://sequencer-nitro-0.sequencer-nitro:9642,ws://sequencer-nitro-1.sequencer-nitro:9642,ws://sequencer-nitro-2.sequencer-nitro:9642
```

Key configuration parameters:
- `replicaCount`: Number of relay replicas to deploy (recommend at least 2 for high availability)
- `configmap.data.chain.id`: Your chain ID
- `configmap.data.node.feed.input.url`: Comma-separated list of WebSocket URLs for all sequencer feed outputs. This relies on the `perReplicaHeadlessService.enabled=true` parameter in the sequencer deployment to create individual services for each sequencer replica.

#### External relays

- Connect to Sequencer Relays (not directly to sequencers)
- Handle all public feed requests
- Provide an additional layer of isolation for production sequencers
- Since these are public facing, ensure they scale appropriately based on your traffic needs:

> ```console

Reviewer: ```suggestion
```shell
```

helm install external-relay offchainlabs/relay \
  --set replicaCount=2 \
  --set configmap.data.chain.id=<your-chain-id> \
  --set configmap.data.node.feed.input.url=ws://sequencer-relay:9642
```

You can check [run a feed relay](/run-arbitrum-node/sequencer/run-feed-relay) to see how to set up a relay node.

### 3. Nitro full node setup

```console
helm install fullnode offchainlabs/nitro \
  --set replicaCount=2 \
  --set configmap.data.parent-chain.id=<parent-chain-id> \
  --set configmap.data.parent-chain.connection.url=<parent-node-url> \
  --set configmap.data.chain.id=<child-chain-id> \
  --set configmap.data.execution.forwarding-target=http://sequencer-nitro:8547
```

#### Send to active configuration (optional)

To enable Redis-based active sequencer discovery:

- Monitor Redis to identify the active Sequencer
- Enable with: `-execution.forwarder.redis-url=redis://<redis-url>:6379`
- Ensure connectivity to individual sequencer services and Redis
- Test failover scenarios before production deployment

> ```console

Reviewer: ```suggestion
```shell
```

helm install fullnode offchainlabs/nitro \
  --set replicaCount=2 \
  --set configmap.data.parent-chain.id=<parent-chain-id> \
  --set configmap.data.parent-chain.connection.url=<parent-node-url> \
  --set configmap.data.chain.id=<child-chain-id> \
  --set configmap.data.execution.sequencer.forwarder.redis-url=redis://<redis-url>:6379
```

#### Mutating-only endpoint (optional)

> For high availability, it is recommended to route mutating transactions to a fleet of `precheckers` which will then forward them to the sequencer. This can be done by setting up a separate endpoint for mutating transactions, and only allowing calls such as `eth_sendRawTransaction` to be routed to the precheckers which then route to the sequencer. This insulates the sequencer from unnecessary load and allows it to focus on transaction ordering. However, this configuration requires custom load balancing logic and is out of scope for this guide.

Reviewer: ```suggestion
For high availability, it is recommended to route mutating transactions to a fleet of `precheckers`, which will forward them to the sequencer. This can be done by setting up a separate endpoint for mutating transactions and only allowing calls such as `eth_sendRawTransaction` to be routed to the precheckers, which then route to the sequencer. This insulates the sequencer from unnecessary load and enables it to focus on transaction ordering. However, this configuration requires custom load balancing logic and is out of scope for this guide.
```

### 4. Redis setup

Set up a highly available Redis cluster for sequencer coordination:

> - **Deployment options**:

Reviewer: ```suggestion
#### Deployment options
```

  - Use a managed service like AWS ElastiCache (recommended)
  - Deploy within Kubernetes using a StatefulSet with PersistentVolumeClaims

> - **Requirements**:

Reviewer: ```suggestion
#### Requirements
```

### 5. Sequencer setup

Deploy multiple sequencer replicas with availability zone spread using the nitro helm chart(availability zone spread is not demonstrated in the example below):

> ```console

Reviewer: ```suggestion
```shell
```

### 7. Batchposter setup

Deploy the batch poster using the nitro helm chart:

> ```console

Reviewer: ```suggestion
```shell
```
